### ISSUE TITLE
TfsAuthenticator hangs the IDE

### ISSUE THREAD

**Issue by mmajcica** [Microsoft/vso-intellij#49]
With the version 1.111.0 of the plugin, I do see a strange behavior of the plugin. In case I do use VSTS Git source control in which I do authenticate on TFS, and then later I attempt to use TFVC, in case the authentication was already provided, IDEA IDE gets unresponsive and I do find following in my log:

```
2016-12-27 12:36:54,227 [  26081]   WARN - fsTelemetryInstrumentationInfo - Unable to locate property resource /telemetry/com.microsoft.alm.plugin-telemetry.properties 
2016-12-27 12:36:54,305 [  26159]   WARN - n.settings.TeamServicesSecrets - getServerContextSecrets: info was null for key:  
2016-12-27 12:36:54,414 [  26268]   WARN - n.settings.TeamServicesSecrets - getServerContextSecrets: info was null for key:  
2016-12-27 12:36:54,492 [  26346]   INFO - .checkout.VsoCheckoutPageModel - Skipping loading contexts in constructor 
2016-12-27 12:36:54,492 [  26346]   INFO - n.context.ServerContextManager - Prompting for credentials 
2016-12-27 12:36:54,492 [  26346]   INFO - ugin.authentication.AuthHelper - getAuthenticationInfoSynchronously calling authenticateAsync on https://tfs.mytfs.com/tfs 
2016-12-27 12:36:54,492 [  26346]   INFO - tion.TfsAuthenticationProvider - starting TfsAuthenticator 
```

In order to get it working I need remove the ServerContextState node from vsts_settings.xml file so that the plugin can request the auth again. Then it works. Same is valid for the vice versa approach, if you initially choose TFVC then try to open VSTS Git source control; TfsAuthenticator Async authentication refuses to start and hangs IDEA.

Regards,
Mario

**Comment by leantk** [comment]
Hi @mmajcica,

Thanks for reporting the issue. There must be a problem where we reuse the login info from the Git repo for the TFVC repo since they share the same account. I'm taking a look at it and will see if I can reproduce the issue on my side. So just to clarify, you clone a Git repo and it works like expected but when you then go to clone a TFVC repo afterwards it hangs. At what point does it hang? Do you see the Checkout dialog again or do you not even get that far?

Thanks again,
-Leah

**Comment by mmajcica** [comment]
Hi @leantk,

indeed. If I do first use Git and login, clone a repo and such. Next time I do try to select the TFVC from the main screen:

![2016-12-23 08_34_54-welcome to intellij idea](https://cloud.githubusercontent.com/assets/3026551/21543548/c81b829e-cdc7-11e6-9fa4-86f23628b16e.png)

Then the app gets stuck, I do NOT get to even see any other window.

The same goes vice versa in case I use the TFVC then try git.

**Comment by mmajcica** [comment]
@volnoboy 

I already mentioned a fix above in my first post:
In order to get it working I need remove the ServerContextState node from vsts_settings.xml file.

Once you remove all ServerContextState nodes from vsts_settings.xml file and restart IDEA, you will get prompted to authenticate again and it will work. Not an elegant way but it works.

Cheers,
Mario

**Comment by volnoboy** [comment]
I cannot locate anywhere this file vsts_settings.xml?

**Comment by mmajcica** [comment]
I have no pc access right now, however it should be in your user folder
under .idea folder somewhere (name of the folder may be different however
I'm sure it starts with a dot). Search for it.

Mario

**Comment by mmajcica** [comment]
@volnoboy 

If you still haven't found it (which I doubt :) ) here is the path in my case:

C:\Users\majcicam\.IdeaIC2016.3\config\options

where majcicam is my user profile folder and .IdeaIC2016.3 is specific to edition and version of IDEA I'm using. Adapt it for your case.

Mario

**Comment by volnoboy** [comment]
Thank you. Now, I found it. I was looking into .idea in my project folder.

**Comment by leantk** [comment]
@volnoboy- we are still looking at fixing this bug permanently so you don't have this pain. We apologize. I will keep this issue updated on the progress. Thanks for your patience.

@mmajcica- thanks for pointing him to the workaround and the right path for the settings file. I'm still debugging the issue at the moment.

**Comment by leantk** [comment]
@mmajcica or @volnoboy,

I made a code change that fixes the issue for me but I couldn't get a true repro of the freeze so I was hoping maybe one of you could test it out for me if you are still having issues. I've attached the updated plugin build here. Please let me know if you see any change or the same thing.

[plugin.idea-1.111.1-freezeFix.zip](https://github.com/Microsoft/vso-intellij/files/687862/plugin.idea-1.111.1-freezeFix.zip)

Thanks for the help,
-Leah

**Comment by mmajcica** [comment]
@leantk 

I just tried it on two machines, different test cases and it now works as expected.
In which version of the plugin will this fix end up?

Thanks

**Comment by leantk** [comment]
@mmajcica it will be version 1.111.1 and I'm looking to release it next week. I'll update this issue once it's out there to download.

**Comment by leantk** [comment]
I just released the fix under the version 1.111.1. Please reopen this issue if you see this behavior again but for now I'm going to close it since I've had 2 people verify it fixed the issue.
